### ISSUE TITLE
 Example app Build failure 

### ISSUE THREAD

**Issue by zzzmx-josh** [prisms-center/phaseField#199]
Hi there,
I am trying to build PRISMS-PF  Example App on  MacbookPro M1, but I encouter  the follwing problems:                                               16 warnings generated.
make[6]: *** No rule to make target `/usr/lib/libdl.dylib', needed by `main'.  Stop.
make[5]: *** [CMakeFiles/main.dir/all] Error 2
make[4]: *** [all] Error 2
make[3]: *** [CMakeFiles/debug] Error 2
make[2]: *** [CMakeFiles/debug.dir/all] Error 2
make[1]: *** [CMakeFiles/debug.dir/rule] Error 2
make: *** [debug] Error 2
I try to make a link of libdl.dylib  into  /usr/lib directory, but it fails again：
ln: /usr/lib/libdl.dylib: Read-only file system
How can I solve them ?  
Many thanks in advance!

**Comment by david-montiel-t** [comment]
Hello, zzzmx_josh

Did you successfully (beforehand) compile the core library of PRISMS-PF?

Can you please list the steps you did after you cloned the repository?

Also, can you try running and compiling a deal.II example beforehand, like
step-1 (this may be a a deal.II issue)?

Best,

David

**Comment by zzzmx-josh** [comment]
I have followed the steps listed in https://prisms-center.github.io/phaseField/doxygen_files/install.html and the core library of  PRISMS-PF has been successfully compiled as follows:
bash-3.2$ cmake .
-- Using the deal.II-9.2.0 installation found at /Applications/deal.II.app/Contents/Resources/Libraries
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_add_test.cmake
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_initialize_cached_variables.cmake
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_invoke_autopilot.cmake
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_pickup_tests.cmake
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_query_git_information.cmake
-- Include macro /Applications/deal.II.app/Contents/Resources/Libraries/share/deal.II/macros/macro_deal_ii_setup_target.cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /Applications/phaseField
bash-3.2$ make -j 6
[ 50%] Built target prisms_pf
[100%] Built target prisms_pf_debug

**Comment by david-montiel-t** [comment]
Thank you.

Can you also list the steps you did to compile and run the application?

David

**Comment by zzzmx-josh** [comment]
1. I directly installed cmake and deal.II from dmg binary packages. The I opened deal.II app to enterl shell to activate the environment.
2.  compile PRISMS-PF core code ：
  $ git clone https://github.com/prisms-center/phaseField.git
  $ cd phaseField
  $ cmake .
  $ make -j 8
3. compile the axample app
  $ cd applications/allenCahn/
  $ cmake .
  $ make -j 6
Then，the problems above appeared:
make[6]: *** No rule to make target /usr/lib/libdl.dylib', needed by main'.  Stop.

**Comment by david-montiel-t** [comment]
Hello, zzzmx_josh

What is your Mac OS version? It looks like there is an incompatibility with
deal.II 9.2.0 and Big Sur. See this thread:

https://github.com/dealii/dealii/issues/10917

As the thread mentions, the issue should be fixed if you download the dmg
file 9.2.0-v3. Can you try that?

If that does not work, can you try the following version (configured to
work with with M1 processors):

https://github.com/dealii/dealii/releases/download/v9.4.0/dealii-9.4.0.ventura-m1.dmg

Finally, you can try installing deal.II using candi, following the approach
described below:

https://github.com/dealii/dealii/wiki/Apple-ARM-M1-OSX

For this option, make sure you clone the file:
https://github.com/dealii/candi/tree/dealii-9.4
(PRISMS-PF is still not compatible with the latest deal.II)

Also, once you have cloned the repository, you can edit the file candi.cfg
and disable all packages except p4est end dealii, which are the only ones
needed by PRISMS-PF.

I hope one of the above options works for you. Let me know if you still run
into any issues.

David

**Comment by zzzmx-josh** [comment]
Hello, David
My Mac OS version is Monterey 12.5.1.
I encountered similar issues when installing deal.II 9.2.0-v3 and dealii-9.4.0.ventura-m1.dmg, where xxx /bin/mpic++ does not exist：

    /Applications/deal.II.app/Contents/Resources/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.3-apple/openmpi-3.1.6-ch3xlssbdriiio4hygk3xhnxclncjvze/bin/mpic++

  is not a full path to an existing compiler tool.

When I finally installed it from candi, I also encountered an error with the following error message:

/Applications/deal.II.app/Contents/Resources/Libraries/include/ArborX/geometry/ArborX_HyperPoint.hpp:61:53: error: expected '(' for function-style cast or type construction
  static_assert(std::is_same_v<FloatingPoint, float>);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/Applications/deal.II.app/Contents/Resources/Libraries/include/ArborX/geometry/ArborX_HyperPoint.hpp:66:1: error: C++ requires a type specifier for all declarations
Point(T x, T y, T z) -> Point<3, T>;
^
5 errors generated.
make[2]: *** [source/arborx/CMakeFiles/obj_arborx_release.dir/access_traits.cc.o] Error 1
make[1]: *** [source/arborx/CMakeFiles/obj_arborx_release.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

 。。。。。
make: *** [all] Error 2
Failure with exit status: 2
Exit message: There was a problem building dealii v9.4.2.

**Comment by david-montiel-t** [comment]
Hello, Josh

Did you disable all dependencies except dealii and p4est when you installed
the candi version?

I think it is probably a good idea to meet via Zoom to go through your
installation issues.

I am available to meet today any time today or tomorrow from 1pm to 6pm US
Eastern Daylight Time.

Are any of these times convenient for you?

Best,

David

**Comment by david-montiel-t** [comment]
Otherwise, please let me know of convenient times for you this week.

David

**Comment by zzzmx-josh** [comment]
Dear David,
I managed to compile the PRISMS application successfully. 
I did this by disabling all dependencies except dealii and p4est and configuring the CMAKE_CXX_COMPILER to use the mpic++ I installed in /opt/brew/bin/mpic++.
One thing I overlooked was forgetting to delete the old CMakeFiles directory from a previous failed compilation. This directory still had the old compilation settings, which caused some errors in my recent attempt.
Thanks for your patience and help throughout this process.
Best,
Josh
